src-cpp/RdKafka.cpp

```diff
@@ -50,3 +50,12 @@ int RdKafka::wait_destroyed (int timeout_ms) {
   return rd_kafka_wait_destroyed(timeout_ms);
 }
 
+void *RdKafka::mem_malloc (void *handle, size_t size) {
```

Comment:
Use RdKafka::Handle *handle as type, not void.

Reply:
Ok, but then you don't mind if I forward declare "Handle" in the header.

Reply:
Maybe it is better with a mem_malloc() method on the Handle class, and one without a handle in the RdKafka namespace.

Reply:
Without a handle completely? So we'd have:
void *RdKafka::mem_malloc(size_t size) and
void *mem_malloc(RdKafka:Handle *handle, size_t size)?
The C api takes a handle (unused at the moment, but still)

Reply:
Yep, that was the idea.

Reply:
Actually I can rearrange a bit I think so there would be no need for two allocators

Reply:
One change though I think I should make, let it accept a NULL handle...

Reply:
My point is that there should be malloc on the Handle, and one without a handle in the global namespace, covering both scenarios. We should avoid NULL-as-optional in C++ if we can.

Reply:
Ok, updated with two methods (with and w/o handle)